### PR TITLE
Fix Build failures:  testAnki2Updates and testTabOrder

### LINKED ISSUE #7467: CI Failure: testAnki2Updates
Failure on all unit tests

```
com.ichi2.anki.tests.libanki.ImportTest > testAnki2Updates[test(AVD) - 4.1.2] FAILED 
	java.lang.AssertionError: expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
```

### LINKED ISSUE #7470: Flaky Test: NoteEditorTest - testTabOrder
```
com.ichi2.anki.NoteEditorTest > testTabOrder[test(AVD) - 7.1.1] FAILED 
	java.lang.AssertionError: Activity never becomes requested state "[DESTROYED]" (last lifecycle transition = "PAUSED")
	at androidx.test.core.app.ActivityScenario.waitForActivityToBecomeAnyOf(ActivityScenario.java:301)
```

Introduced in #7436 

Test should not be running - likely needs a rule to ignore

### PR DESCRIPTION
**Pull request by david-allison-1**
IDE-based lint warning on comparing using == rather than `.equals()`

Please check 9674085 when reviewing - it appeared incorrect, but maybe I missed the context of the code.

## Fixes
Fixes #7467 
Fixes #7470

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/utils/ObjectUtils.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+package com.ichi2.libanki.utils;
+
+public class ObjectUtils {
+    private ObjectUtils() {
+
+    }
+
+    public static boolean equals(Object a, Object b) {
+        // Normally we'd use compat - but we're optimising for perf.
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/libanki/utils/ObjectUtils.java:25]: `Utils.equals` seems to be what you are looking for here

**Review by Arthur-Milchior — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/libanki/utils/ObjectUtils.java (hunk 1 above)
Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/libanki/utils/ObjectUtils.java:25]: I couldn't find it - thanks!

**Review by Arthur-Milchior — APPROVED** [review]

**Comment by david-allison-1** [comment]
Added a fix for #7470 to ensure build is green

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorTabOrderTest.java [hunk 2]
```diff
@@ -44,21 +45,22 @@
...
-        intent.putExtra(NoteEditor.EXTRA_CALLER, NoteEditor.CALLER_DECKPICKER);
-        return intent;
+public class NoteEditorTabOrderTest {
+
+    public NoteEditorTabOrderTest() {
+        // Rules mean that we get a failure on API 25.
+        // Even if we ignore the tests, the rules cause a failure.
+        // We can't ignore the test in @BeforeClass ("Test run failed to complete. Expected 150 tests, received 149")
+        // and @Before executes after the rule.
+        // So, disable the rules in the constructor, and ignore in before.
```
Comment (mikehardy) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/NoteEditorTabOrderTest.java:55]: Wellll....that's unexpected. Wow.

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.java [hunk 3]
```diff
@@ -408,7 +408,7 @@ private void _prepareModels() {
             if (srcScm.equals(dstScm)) {
                 // they do; we can reuse this mid
                 Model model = srcModel.deepClone();
-                model.put("id", mid);
+                model.put("id", (long) mid);
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.java:411]: interesting! I guess that was a confluence of your change at the same time as the "less long" PR

**Review by mikehardy — APPROVED** [review]
This is some subtle stuff right there...not sure what the lesson learned here is other than velocity results in change and *any* change can result in unexpected failures. So in a sense this is unexpected (though I guess almost always to be expected in complicated Andoid projects...) janitor work - thanks very much for taking out the trash

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.java (hunk 3 above)
Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.java:411]: I didn't touch this line - came about from the conversion to Long, one of the two branches was converted.

**Pull request merged**